ogr/ogrutils.cpp

```diff
@@ -1364,16 +1363,19 @@ char* OGRGetXMLDateTime(const OGRField* psField, bool bAlwaysMillisecond)
                      (TZFlag > 100) ? '+' : '-', TZHour, TZMinute);
     }
 
+    // sizeof() includes null terminator. +6 is to make -Wformat-truncation= happy
+    constexpr size_t nMaxSize = sizeof("YYYY-MM-DDThh:mm:ss.sss+hh:mm")+6;
```

Comment:
Possibly stupid consideration, unrelated to this PR: Do you think we'll ever encounter a "YYYY-MM-DDThh:mm:ss.sss+-03:00" format string in the wild (with '+-' literal)? (Eg. if some external app which is generating the strings has the "+" character hardcoded in their string format and then feeds in a negative offset value)

Reply:
> Possibly stupid consideration, unrelated to this PR: Do you think we'll ever encounter a "YYYY-MM-DDThh:mm:ss.sss+-03:00" format string in the wild (with '+-' literal)?

I don't see a compelling reason to take into account this situation, at least not until proven that there's a major data source that would have such bug in its data